### ISSUE TITLE
[SDK 2.1.300] Command 'dotnet publish' fails for projects with GeneratePackageOnBuild=true

### ISSUE THREAD

**Issue by gitorrin** [dotnet/core#1778]
Command 'dotnet publish' fails for projects publishing with GeneratePackageOnBuild=true. 
TargetFramework is netcoreapp2.0.

`<PropertyGroup>
    <TargetFramework>netcoreapp2.0</TargetFramework>
    <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
  </PropertyGroup>`

This used to work with the previous SDK version 2.1.101. 
The error it generates is:
"C:\Program Files\dotnet\sdk\2.1.300\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Publish.targets(168,5): error MSB3030: Could not copy the file "obj\Debug\netcoreapp2.0\<anyAssemblyName>.dll" because it was not found. "

Workaround: build first, and then publish.

**Comment by dasMulli** [comment]
Do you have ay steps to reproduce? what project type is affected? I couldn't reproduce with a couple of configurations I thought you could have.

**Comment by gitorrin** [comment]
Sorry, apparently I haven't added enough details. Thank you for reacting so fast. 

1. Create new VS solution. 
2. Add new Net Core project (ClassLibrary1), type class library (make sure csproj shows TargetFramework =netcoreapp2.0)
3. Set the project property GeneratePackageOnBuild=true (csproj)

csproj file content:
`<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>netcoreapp2.0</TargetFramework>
    <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
  </PropertyGroup>
</Project>`

4. Add global.json to project folder with content 
`{
  "sdk": {
    "version": "2.1.300"
  }  
}
`
4. Go to command line and run dotnet publish

Fails with error 

> C:\Program Files\dotnet\sdk\2.1.300\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Publish.targets(168,5): error MSB3030: Could not copy the file "obj\Debug\netcoreapp2.0\ClassLibrary1.dll" because it was not found. [C:\Users\<user>\Documents\Visual Studio 2017\Projects\NetPublishIssue\ClassLibrary1\ClassLibrary1.csproj]

**Comment by dasMulli** [comment]
Thanks, can repro now. the important part to the repro is that no `dotnet build` etc. have been called before.

**Comment by dasMulli** [comment]
@nguerrera the problem here is that the new `--no-build` option for publish is interfering with the existing NuGet logic for `GeneratePackageOnBuild` which has logic to not trigger a build again while packing.

In essence, both pieces of logic use `$(NoBuild)` for this 😱

**Comment by dasMulli** [comment]
@gitorrin you are right, this is a regression in 2.1.3xx. The workaround is to explicitly build (`dotnet build`) before publishing.

**Comment by gitorrin** [comment]
Thanks for looking into this, @dasMulli.

**Comment by Petermarcu** [comment]
Issue moved to [dotnet/cli #9656](https://github.com/dotnet/cli/issues/9656) via [**ZenHub**](https://www.zenhub.com/)
